scripts/e2e_tests.sh

```diff
@@ -0,0 +1,150 @@
+            echo -e "      \t\t  Example: ./scripts/e2e_tests.sh -a x86 -e send"
+            echo -e "\n  -a\t\tSelect ABI (required)"
+            echo -e "      \t\t  Options:"
+            echo -e "      \t\t  'x86_64' - default system image: API 30 google_apis_playstore x86_64"
+            echo -e "      \t\t  'x86' - default system image: API 30 google_apis_playstore x86"
+            echo -e "      \t\t  'arm64-v8a' - default system image: API 30 google_apis_playstore x86_64"
+            echo -e "      \t\t  'armeabi-v7a' - default system image: API 30 google_apis_playstore x86"
+            echo -e "\n  -l\t\tSelect API level (optional)"
+            echo -e "      \t\t  Minimum API level: 23"
+            echo -e "\n  -t\t\tSelect API target (optional)"
+            echo -e "      \t\t  See examples on selecting system images below"
+            echo -e "\n  -s\t\tCreate an AVD and snapshot for quick-boot (optional)"
+            echo -e "      \t\t  Does not run app"
+            echo -e "\n  -x\t\tSet timeout in seconds for emulator launch and AVD boot-up (optional)"
+            echo -e "      \t\t  Default: 1800"
+            echo -e "      \t\t  Must be an integer"
+            echo -e "\nExamples:"
+            echo -e "  '$(basename $0) -a x86_64 -s'\tCreates an AVD and quick-boot snapshot for x86_64 ABI"
+            echo -e "  '$(basename $0) -a x86_64'   \tEmulates app for x86_64 ABI from snapshot"
+            echo -e "  '$(basename $0) -a x86 -l 29 -t google_apis'"
+            echo -e "                             \t\tSelect system image \"system-images;android-29;google_apis;x86\""
+            echo -e "\nRecommended system images for testing ARM ABIs:"
+            echo -e "  armeabi-v7a:"
+            echo -e "    \"system-images;android-30;google_apis_playstore;x86\" - default"
+            echo -e "    \"system-images;android-30;google-tv;x86\""
+            # TODO: add list of supported images for arm64-v8a
+            echo -e "\nFor a full list of system images run 'sdkmanager --list'"
+            exit 1
+            ;;
+        ?)
+            echo "Try '$(basename $0) -h' for more information." >&2
+            exit 1
+            ;;
+    esac
+done
+
+#setting variables
+if [ ! -d "./android/app" ]; then
+    echo "Error: Incorrect working directory" >&2
+    echo "Try './scripts/$(basename $0)' from zingo-mobile root directory." >&2
+    exit 1
+fi
+
+test_report_dir="app/build/outputs/emulate_app_reports/${abi}"
+rm -rf "${test_report_dir}"
+mkdir -p "${test_report_dir}"
+
+avd_name="${avd_skin}-android-${api_level}_${api_target}_${arch}"
+sdk="system-images;android-${api_level};${api_target};${arch}"
+platform="platforms;android-${api_level}"
+
+# "preparing emulator"
+if ! (adb devices | grep emulator-5554) || "${reboot_emulator}"; then
+  emulate_command="./scripts/emulate_app.sh -a ${abi} -l ${api_level} -t ${api_target}${create_snapshot_option} -x ${timeout_seconds} -Y"
```

Comment:
oh ok ive got to this part and ive realised whats going on here...
so can we not have this whole script just be:
1. setup/create avd -> emulate_app.sh -s blah blah blah
2. boot emulator -> emulate_app.sh -Y blah blah blah (using your non-boot-app changes)
3. build detox and prep -> kill all node, start yarn react-native, yarn detox build etc.
4. run e2e tests -> yarn detox test etc.
5. kill emulator when done -> some logic to trigger `kill_emulators.sh`

i think there is lots of unnecessary complications and code that is already handled by the `emulate_app.sh` script, like creating test directories, specifying which ABI etc. 

if anything else is needed for e2e why not add them to the `emulate_app.sh` itself?
maybe we could also simplify the `integraton tests.sh` to use the `emulate_app.sh` further down the line too so there isnt so much duplication of code

Reply:
youre right, theres a better way to do this